src/rgw/driver/sfs/types.cc

```diff
@@ -93,6 +100,63 @@ Object* Object::create_commit_new_object(const rgw_obj_key& key, SFStore* store,
   return result;
 }
 
+Object* Object::try_create_with_last_version_from_database_fetch(
+    SFStore* store, const std::string& name, const std::string& bucket_id) {
+  sqlite::SQLiteObjects objs(store->db_conn);
+  auto obj = objs.get_object(bucket_id, name);
+  if (!obj) {
+    return nullptr;
+  }
+
+  sqlite::SQLiteVersionedObjects objs_versions(store->db_conn);
+  auto last_version = objs_versions.get_last_versioned_object(obj->uuid);
+  if (!last_version.has_value()) {
+    return nullptr;
+  }
```

Comment:
is it possible the object underneath us will change in-between these two calls to the database? E.g., say the object is changed between obtaining it from the bucket (i.e., `obj`) and obtaining the last version (i.e., `last_version`). Given we are relying on both for the resulting object below, is it not possible that we will be returning a version that does not meet the expected result in `meta`?

Reply:
There is something here that doesn't add up.

Having a combined query would be better, but why does object have the same info as versioned object?

Is this actually right data for result->meta ? This is mostly in spirit of the previous code, but wondering if this shouldn't be

```
  result->meta = {
    .size = last_version->size,
    .etag = last_version->etag,
    .mtime = last_version->creation_time,
    .set_mtime = obj->set_mtime,
    .delete_at = last_version->deletion_time
  };
```

Paging @0xavi0

Reply:
This was discussed time ago....
object should only store the common info:
`uuid`, `name` and `bucket_id` 
The rest of info should be picked from the version.

As this is being around for long... I've created an issue for it, (https://github.com/aquarist-labs/s3gw/issues/395) so we can address it.